src/book/03-guides/04-general/01-git.mdx

```diff
@@ -64,6 +66,28 @@ If your code is in a different repository and it uses C++ or protobuf files, use
 
 If you push your changes to GitHub and get a message that the build failed due to formatting issues, try running the relevant format commands, commit any changes, and push to GitHub again. This will trigger a new build, which will check the formatting again. You won't be able to merge the PR until the build checks pass.
 
+## Reviewing Code
+
+To review code on GitHub open up a PR and open the Files Changed tab. This will give you the diff of the PR. You can click on the lines of code in the diff to add a comment and/or a suggestion to those lines. You can either leave a single comment or start a review. In a review all comments you make will have pending written on them to say that they have not been posted yet. Please make each comment non-ambiguous and clearly state what your concern is.
+
+When reading the code, if there is anything that confuses you or think is incorrect make a comment with all your concerns. For example, if there is a nontrivial block of code without a comment explaining it you should ask for the author to provide such a comment. Another example is if there is a block of code that could do something that the autor may have not considered you should inform the author, with a comment, on what these other effects may be.
+
+When finished reading the code and you have a bunch of pending comments for your review you can do one of three things. The first is to request changes, this blocks the code from being merged until you approve the branch, even if others have approved it. The second is to approve the PR, this allows it to be merged into master, don't do this unless you are confident that the code is correct. The third is to leave a comment, do this if you are not too concerened about the comments you have made.
```

Comment:
```suggestion
After reading the code and adding your pending comments, click **Finish your review** at the top right. This will show a popup with an input for overall comments, as well as the three actions you can take to finish your review:

- **Request changes**: this blocks the code from being merged until you subsequently approve the PR in another review, even if others have approved it.
- **Approve**: this allows the PR to be merged, so don't do this unless you are confident that the changes are correct.
- **Comment**: this adds your comments without blocking or approving the changes. Do this if you are not too concerned about the comments you have made.
```